### PR TITLE
Integrate Facebook Ads into your Android application in Android Studio

### LINKED ISSUE #4585: [API] Integrating Facebook Ads with your Android Application in Android studio
#### Introduction
Besides developing mobile applications for technological adjustments, developers always look towards getting something in return. One of the greatest ways of earning from applications being advertisement, this article shows developers how they can integrate banner and interstitial ads in their application.

With Facebook Audience Network, developers are allowed to monetize their android applications with Facebook Ads. This motivates developers to come up with new applications every now and then.

#### Key takeaways 
- Overview of all the requirements
- Set up your Android application project
- Installation of the required **sdk** i.e **(Facebook Network Audience) sdk** 
       and other requirements.
- Steps of **Integrating**FAN sdk and adapter into your Android application

#### Article quality
The official documentation gives a general directive using the initial generated code, which sometimes proves hard to follow.
I intend to give a clear step by step explanation with more understanding simplicity and flexibility with different versions (not offered in the available documentation).

#### References
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by del-ui**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4585 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by del-ui** [comment]
I have made all the necessary changes @hectorkambow 
cc: @Jethro-magaji

**Comment by del-ui** [comment]
@jerimkaura I think the article had already gone past plagiarism check stage

**Comment by Jethro-magaji** [comment]
@del-ui 
Have you made the necessary changes i requested in the last PR and made the corrections @hectorkambow suggested?
Please make sure everything is sorted out, so that i can continue the review process.
Thanks.

**Comment by del-ui** [comment]
Yeah, everything is in place @Jethro-magaji

**Comment by jerimkaura** [comment]
> @jerimkaura I thought the article had already gone past the plagiarism check stage

We do that to make sure that we don't miss anything, as a new PR, You might have added new content that needs to be verified for originality.

Not a problem if you content is original.

**Comment by del-ui** [comment]
> > @jerimkaura I thought the article had already gone past the plagiarism check stage
> 
> We do that to make sure that we don't miss anything, as a new PR, You might have added new content that needs to be verified for originality.
> 
> Not a problem if you content is original.

Okay

**Comment by Jethro-magaji** [comment]
@ahmadmardeni1 
I would like to continue the review process, could you please reassign me to this PR, as it was a duplicate.
Thanks

**Comment by Jethro-magaji** [comment]
> Yeah, everything is in place @Jethro-magaji

Alright.

**Comment by LinusMuema** [comment]
plagiarism check done :heavy_check_mark:

File: content/articles/integrate-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+- Facebook business account
+
+You also need to be familiar with the layout and how to use Android studio. With all these requirements in place, you can move on to the steps involved in the integration process.
+
+Assuming you have your Android studio up and running, the only thing you have to install is the FAN SDK.
+
+ **FAN SDK installation**
+ To install, you can download your SDK manually from [here](https://developers.facebook.com/resources/audience-network-sdk-6.8.0.zip) (Not recommended) or follow the steps in [Stage 1 integration steps](#stage-1-integration-steps) to install it the right way.
+
+### Stage 1 Integration steps
```
Comment (Jethro-magaji) [inline content/articles/integrate-facebook-ads-into-your-android-application-in-android-studio/index.md:43]: Where's stage 2 and 3?

File: content/articles/integrate-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+- Facebook Audience Network SDK
+- Facebook Audience Network account
+- Android studio
+- Facebook business account
+
+You also need to be familiar with the layout and how to use Android studio. With all these requirements in place, you can move on to the steps involved in the integration process.
+
+Assuming you have your Android studio up and running, the only thing you have to install is the FAN SDK.
+
+ **FAN SDK installation**
```
Comment (Jethro-magaji) [inline content/articles/integrate-facebook-ads-into-your-android-application-in-android-studio/index.md:40]: Please write the full meaning of "FAN" too.

**Review by Jethro-magaji — COMMENTED** [review]
Hello @del-ui 
I have gone through your article and made some reviews, please run your article over [Grammarly ](https://grammarly.com/) to correct some grammatical errors. You can also download the Grammarly Chrome extension to make your work easy.

**Comment by hectorkambow** [comment]
@WanjaMIKE  

Lets keep an eye on this PR and assist Jethro with anything that he may need for this test review. 

This will be the final one (unless otherwise needed) before we get him set up with the GitHub permissions. 🚀

**Comment by hectorkambow** [comment]
cc @section-engineering-education/content-moderator-team

**Comment by WanjaMIKE** [comment]
Noted

**Comment by del-ui** [comment]
@Jethro-magaji you can check whether the changes I made are as you requested.

File: content/articles/integrate-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 3]
```diff
@@ -30,15 +30,14 @@ Monetizing Applications counts as one of the success of a developer other than p
...
+- [Android studio](#https://developer.android.com/studio/releases)
 - Facebook business account
 
 You also need to be familiar with the layout and how to use Android studio. With all these requirements in place, you can move on to the steps involved in the integration process.
-
 Assuming you have your Android studio up and running, the only thing you have to install is the FAN SDK.
 
  **FAN (Facebook Audience Network) SDK installation**
- To install, you can download your SDK manually from [here](https://developers.facebook.com/resources/audience-network-sdk-6.8.0.zip) (Not recommended) or follow the steps in [Stage 1 integration steps](#stage-1-integration-steps) to install it the right way.
+ To install, you can download your SDK manually from [here](https://developers.facebook.com/resources/audience-network-sdk-6.8.0.zip) (Not recommended) or follow the steps in [Initial integration steps](#initia-integration-steps) to install it the right way.
```
Comment (Jethro-magaji) [inline content/articles/integrate-facebook-ads-into-your-android-application-in-android-studio/index.md:40]: (#initia-integration-steps); the "initia" spelling isn't correct.

File: content/articles/integrate-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 4]
```diff
@@ -72,10 +72,10 @@ Once done, you are ready for the integration process in the Android studio.
 
 ### Integration in Android Studio
 
-Don't have an Application? No need to worry since you can set-up an simple Application in a few steps below.
+Don't have an Application? No need to worry since you can set up a simple application in a few steps below.
 
 #### Step 1: Create a New Project
-If you don't have an application created already, open your Android Studio and create a new project with Ads activity. Since it is a test Application, you can use it without altering. This [article](https://www.section.io/engineering-education/first-android-App/) can be very useful in helping you with proper project creation.
+If you don't have an application created already, open your Android Studio and create a new project with Ads activity. Since it is a test Application, you can use it without altering it. This [article](https://www.section.io/engineering-education/first-android-App/) can be very useful in helping you with proper project creation.
```
Comment (Jethro-magaji) [inline content/articles/integrate-facebook-ads-into-your-android-application-in-android-studio/index.md:78]: There's no need to include this "If you don't have an application created already" here again, you have already mentioned it above.

File: content/articles/integrate-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 5]
```diff
@@ -23,14 +23,14 @@
 
 Apart from designing mobile applications for technological advancements, developers are always looking for a return on their investment. This article demonstrates developers how to include banner and interstitial Ads in their applications, which is one of the best ways to earn money from apps.
 
-Monetizing Applications counts as one of the success of a developer other than producing the required product. This motivates developers and programmers to come up with more software with a thought in mind that there will be an appreciation token.
+Monetizing Applications counts as one of the successes of a developer other than producing the required product. This motivates developers and programmers to come up with more software with a thought in mind that there will be an appreciation token.
 
 ### Prerequisites
 
 The key requirements for a successful integration include;
 - Facebook Audience Network SDK
```
Comment (Jethro-magaji) [inline content/articles/integrate-facebook-ads-into-your-android-application-in-android-studio/index.md:31]: Please include links for the rest of the list.

File: content/articles/integrate-facebook-ads-into-your-android-application-in-android-studio/index.md [hunk 6]
```diff
@@ -23,14 +23,14 @@
...
 - Facebook Audience Network account
```
Comment (Jethro-magaji) [inline content/articles/integrate-facebook-ads-into-your-android-application-in-android-studio/index.md:32]: Add a link

**Review by Jethro-magaji — COMMENTED** [review]
@del-ui 
Well done for the major edits, however, there are a few edits remaining which I have just reviewed.

**Comment by del-ui** [comment]
All done @Jethro-magaji

**Comment by hectorkambow** [comment]
@Jethro-magaji 
can we get a status on this PR ? 

If any more major revisions are needed - we will be closing this PR asap. 👀 

@del-ui 
Please be sure you are going over the entire article as our peer reviewers are not responsible to point out every mistake - that is the authors responsibility.

**Comment by Jethro-magaji** [comment]
Hi! @hectorkambow 
The article is now okay, there are no further revisions.

I have just concluded my review for this PR, this article is ready for final review 
Cc @LinusMuema

**Comment by LinusMuema** [comment]
Hi @Jethro-magaji 
Once you have completed your review just add the "ready for final review" label and a CM will take care of it.

**Comment by Jethro-magaji** [comment]
> Hi @Jethro-magaji Once you have completed your review just add the "ready for final review" label and a CM will take care of it.

Alright, but i can't add it now because i don't have the permission to do so.
Thanks

**Comment by del-ui** [comment]
@hectorkambow have a look at this

**Comment by hectorkambow** [comment]
@Jethro-magaji Thank you for the review. 

jumping on this one

**Comment by hectorkambow** [comment]
@del-ui thank you taking a look right now!

**Comment by hectorkambow** [comment]
@del-ui 

Does this paragraph seem correct to you ? 


cc @Jethro-magaji  

image

cc @WanjaMIKE

**Comment by hectorkambow** [comment]
@del-ui Victor there are some significant readability challenges in your article that raise concerns over how much review/editing would be required (still) from our peer review team. 

We don't provide editing services and recommend that students leverage online editing services (there are both free and paid services that you can explore), or try engaging with the community or other peers in your network to have them review prior to submitting a revised proposal. 

This is a duplicate article and should have not needed any more revisions when re-submitted.

Thank you.


cc: @Jethro-magaji  this article was not ready for final - it was missing the front matter (see docs) and your peer reviewer line (see my changes towards the bottom). 

An article of this quality should not have the ready for final review tab on it - it should more than likely be closed to allow the author the proper time to make the required edits.

**Comment by hectorkambow** [comment]
image

**Comment by Jethro-magaji** [comment]
> @del-ui Victor there are some significant readability challenges in your article that raise concerns over how much review/editing would be required (still) from our peer review team.
> 
> We don't provide editing services and recommend that students leverage online editing services (there are both free and paid services that you can explore), or try engaging with the community or other peers in your network to have them review prior to submitting a revised proposal.
> 
> This is a duplicate article and should have not needed any more revisions when re-submitted.
> 
> Thank you.
> 
> cc: @Jethro-magaji this article was not ready for final - it was missing the front matter (see docs) and your peer reviewer line (see my changes towards the bottom).
> 
> An article of this quality should not have the ready for final review tab on it - it should more than likely be closed to allow the author the proper time to make the required edits.

Thanks, @hectorkambow for the heads up I will note these things in the next review.

**Comment by del-ui** [comment]
@hectorkambow Could kindly re-open this PR so that I can make some commits instead of re-opening a new one?

cc @Jethro-magaji

**Comment by hectorkambow** [comment]
> @hectorkambow Could kindly re-open this PR so that I can make some commits instead of re-opening a new one?
> 
> 
> 
> cc @Jethro-magaji 

@del-ui 
Sorry we don't allow PRs being open that long (or in the queue while revisions are being done - that should be done in your personal forked repo) the best would be for a new PR to be opened. Thank you.

**Comment by del-ui** [comment]
For that case, you closed the issue for this PR, can I still open a new PR when the issue is closed?

@hectorkambow

**Pull request closed without merging**